internal/commands/update_stemcell.go

```diff
@@ -97,16 +97,19 @@ func (update UpdateStemcell) Execute(args []string) error {
 			update.Logger.Printf("No change for release %q\n", rel.Name)
 			continue
 		}
-
-		local, err := releaseSource.DownloadRelease(update.Options.ReleasesDir, remote)
-		if err != nil {
-			return fmt.Errorf("while downloading release %q, encountered error: %w", rel.Name, err)
-		}
-
 		lock := &kilnfileLock.Releases[i]
-		lock.SHA1 = local.Lock.SHA1
+
 		lock.RemotePath = remote.RemotePath
 		lock.RemoteSource = remote.RemoteSource
+		lock.SHA1 = remote.SHA1
+		if remote.SHA1 == "" || remote.SHA1 == "not-calculated" {
```

Comment:
Where is "not-calculated" from? Maybe pull this out into a const and add a note on what emits this value.

(It was cool to see the PR review request. I miss y'all. Hope you're all well.)

Reply:
The `s3` and `github` sources can set this as  [`not-calculated`](https://github.com/search?q=repo%3Apivotal-cf%2Fkiln%20not-calculated&type=code) particularly when no-download specified.